src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -828,19 +831,19 @@ private static void RestoreViewName(DataSourceEntry ds, string dataSetName)
             }
 
             // Write out of all the other state for roundtripping 
-            string extraContent = (topParentname ?? controlName) + ".editorstate.json";
+            string extraContent = (topParentname ?? newControlName) + ".editorstate.json";
```

Comment:
should this always be `newControlName` for editorState?

Reply:
we always dump the editorstate data per top control, for AppTest control since we are sharding into individual TestSuites the editorState.json would still be 1.